src/make-styles.tsx

```diff
@@ -144,18 +155,19 @@ const matchMediaRule = (mediaRule: string, theme: Theme, windowDimensions: Scale
 const useStyleSheet = (
   styles: NestedStyles,
   theme: Theme,
-  windowDimensions: ScaledSize
+  windowDimensions: ScaledSize,
+  containerDimensions: GenericSize,
 ): { [key: string]: Style } => {
   return useMemo(() => {
     const finalStyles: { [key: string]: Style } = {};
     // we need to make sure to do a deep clone here, so that theme and viewport units can be resolved from original strings
     const stylesCopy = { ...styles };
     for (const key in stylesCopy) {
       const { main, ...mediaStylesCopy } = stylesCopy[key];
-      // this will contain the main style and all applicapble media query styles
+      // this will contain the main style and all applicable media and container query styles
       const mediaStylesArray = [resolveThemeVariables({ ...main }, theme, windowDimensions)];
       for (const mediaRule in mediaStylesCopy) {
-        if (matchMediaRule(mediaRule, theme, windowDimensions)) {
+        if(matchQueryRule(mediaRule, theme, windowDimensions, containerDimensions.height > 0 ? containerDimensions : undefined)) {
```

Comment:
it seems to me like this will always use container dimensions no matter whether the rule is declared as `@media` or `@container` if a container context is present